test/parallel/test-http-request-dont-override-options.js

```diff
@@ -33,13 +34,11 @@ http.createServer(function(req, res) {
   }).end();
```

Comment:
You could just put `server.close()` after `res.resume()` and then validate the `assert` conditions there (lines 37-42). That way we don't need the `process.on('exit')`.